nmdp-bioinformatics/gfe-db
# gfe-db update service

This PR contains basic infrastructure to build and load `gfe-db` into a running Neo4j server:
- AWS Batch Job Definitions, Queues and Compute Environments for build and load services
- StepFunctions workflow to run build and load sequentially
- Uploads logs in addition to CSV dataset
- Tested with and without alignments
- Built and loaded versions 3450, 3440, and 3430
- Refactored repo into 3 basic components each with its own Dockerfile: 
  - Database
  - Build service
  - Load service
- Updated README

To Dos:
- [ ] Use Fargate with AWS Batch for the load service instead of EC2 to save cost
- [ ] Create nested cloudformation templates
- [ ] Add CI/CD for Docker images
- [ ] Add trigger for when a new IMGT/HLA version is released
- [ ] Before uploading CSVs, the S3 prefix should be cleared of previously built data.
- [ ] Load script can be optimized
  - [ ] Better logging
  - [ ] Clean up the Cypher script to avoid Neo4j errors
- [ ] Add a Makefile
- [ ] Add SSL policy for Neo4j to use HTTPS
- [ ] Deploy Neo4j with the APOC and Data Science plugins
- [ ] Update the userdata in the database.yml template to use the current version
- [ ] Add structured logs to the scripts
- [ ] Remove logs, CSVs from S3 prefix before a new build begins
- [ ] Update the Neo4j configuration to set users and roles for security
- [ ] Add architecture diagram to documentation
